### PR TITLE
Added validation for voucher generation field

### PR DESCRIPTION
**Pull request by jfwiebe**
Pressing "Generate random vouchers" will result in a malformed AJAX request and an undefined error message.
image
I've added a check that the input field is not empty and changed the input field to `type=number`.

Please give it a short test before merging.

### CODE REVIEW

File: src/pretix/static/pretixcontrol/js/ui/main.js [hunk 1]
```diff
@@ -96,12 +96,17 @@ $(function () {
...
-        });
+        var num = $("#voucher-bulk-codes-num").val();
+        if (num != "") {
+            var url = $(this).attr("data-rng-url");
+            $("#id_codes").html("Generating...");
+            $.getJSON(url + '?num=' + num, function (data) {
+                $("#id_codes").val(data.codes.join("\n"));
+            });
+        } else {
+            alert("Please enter a number!");
```
Comment (rixx) [inline src/pretix/static/pretixcontrol/js/ui/main.js]: Hm, an alert? Wouldn't a message next to the field be less intrusive?

**Comment by jfwiebe** [comment]
Yeha it would. Currently it gives an alert so I wanted to clarify the message. Maybe also using Bootstrap Validation States (https://getbootstrap.com/css/#forms-control-validation) would look great.

**Comment by jfwiebe** [comment]
@rixx what about this? or too lazy?

**Comment by raphaelm** [comment]
I like it, I think it is clear enough.
One final note: Could you add another ``$(".form-group:has(#voucher-bulk-codes-num)").removeClass("has-error");`` into the first part of the ``if`` block? So that it does not stay red if someone submits something valid :)

**Comment by rixx** [comment]
Looks good to me :)

**Comment by raphaelm** [comment]
Top! Thank you very much!

**Pull request merged**